### DISCUSSION TITLE
How to import a bundle css in AssetMapper

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#57952] [Q&A]
I have a bundle with a stylesheet in assets/styles.css.   

How do I include that asset in AssetMapper?  That is, something like this (which doesn't work)

```bash
bin/console importmap:require ./vendor/survos/my-bundle/assets/style.css 
```

I looked at https://github.com/symfony/symfony/issues/51329, but it's not quite the same.  Or at least I couldn't figure out an answer.

I've exposed the css in package.json, but I'm not sure how to incoporate it, either into bundle twig files or the application that calls the bundle, except my moving it to public.

For now, I'm simply adding the stylesheet to the html, but I keep thinking there must be a way to use assetmapper.

    <link rel="stylesheet" type="text/css" href="{{ asset('bundles/survospixie/style.css') }}" ></link>

**Comment by smnandre** [comment]
This is indeed the best way to do it. 
https://symfony.com/doc/current/frontend/asset_mapper.html#third-party-bundles-custom-asset-paths

AssetMapper can version and deploy any public asset exposed by a bundle.


So if you have a bundle AcmeFooBarBundle.. you can put the stylesheet in the public directory of the bundle (ex: `public/css/theme.css`)

```
├── config
├── public
│   ├── ...
│   ├── css
│   │   └── theme.css
├── src
│   ├── ...
│   └── AcmeFooBarBundle.php
├── templates
├── ...

```


Inside the app where the bundle is used, you need to call the `asset` Twig function, it will automatically handle paths & versionning

```twig
{# templates/blog.html.twig #}

<link rel="stylesheet" href="{{ asset('bundles/acmefoobar/css/theme.css') }}">
```


```html
<!-- will be rendererd in HTML -->
<link rel="stylesheet" href="/assets/bundles/acmefoobar/css/theme-a4b2d42.css">
```

---

> but I keep thinking there must be a way to use assetmapper.

You cannot assume wether or not the app uses AssetMapper, WebpackEncore, or... none of them. So it's best for a bundle to expose public assets and let apps use them the way they can :)
